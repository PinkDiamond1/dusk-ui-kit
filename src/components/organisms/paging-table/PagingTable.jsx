/* @flow */
import * as React from 'react'
import BootstrapTable from 'react-bootstrap-table-next'
import paginationFactory from 'react-bootstrap-table2-paginator'

export const PagingTableVariants = {
  PRIMARY: 'primary',
  SECONDARY: 'secondary',
  INFO: 'info',
  SUCCESS: 'success',
  WARNING: 'warning',
  DANGER: 'danger',
}

type Props = {
  className: string,
  data: array,
  columns: array,
  variant: 'primary' | 'secondary' | 'info' | 'success' | 'warning' | 'danger',
}

const pagingOptions = {
  // pageStartIndex: 0,
  sizePerPage: 6,
  hideSizePerPage: true,
  hidePageListOnlyOnePage: true,
}

const PagingTable = (props: Props): React.Element<*> => {
  const { className, data, columns, variant } = props
  let updatedClassname = `__duk-paging-table ${className}`
  switch (variant) {
    case 'primary':
      updatedClassname += '__duk-paging-table--primary'
      break
    default:
      break
  }
  return (
    <BootstrapTable
      classes={updatedClassname}
      keyField="id"
      data={data}
      columns={columns}
      bootstrap4
      pagination={paginationFactory(pagingOptions)}
      bordered={false}
    />
  )
}

PagingTableVariants.defaultProps = {
  className: '',
  data: [],
  columns: [],
  variant: PagingTableVariants.PRIMARY,
}

export default PagingTable
