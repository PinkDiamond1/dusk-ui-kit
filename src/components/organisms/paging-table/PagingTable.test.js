import React from 'react'
import renderer from 'react-test-renderer'

import PagingTable from './PagingTable'

const data = [
  {
    id: 1,
    name: 'Dusk Chicken',
    price: '$1000',
  },
]
const columns = [
  {
    dataField: 'id',
    text: 'Product ID',
  },
  {
    dataField: 'name',
    text: 'Product Name',
  },
  {
    dataField: 'price',
    text: 'Product Price',
  },
]

it('test to see if the PagingTable renders correctly', () => {
  const tree = renderer.create(<PagingTable data={data} columns={columns} />).toJSON()
  expect(tree).toMatchSnapshot()
})
