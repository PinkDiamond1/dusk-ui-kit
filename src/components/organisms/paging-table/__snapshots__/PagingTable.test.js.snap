// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test to see if the Header renders correctly 1`] = `
Array [
  <div
    className="react-bootstrap-table"
  >
    <table
      className="table __duk-paging-table undefined"
    >
      <thead>
        <tr>
          <th
            tabIndex={0}
          >
            Product ID
          </th>
          <th
            tabIndex={0}
          >
            Product Name
          </th>
          <th
            tabIndex={0}
          >
            Product Price
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          className={null}
          style={Object {}}
        >
          <td>
            1
          </td>
          <td>
            Dusk Chicken
          </td>
          <td>
            $1000
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  <div
    className="row react-bootstrap-table-pagination"
  >
    <div
      className="col-md-6 col-xs-6 col-sm-6 col-lg-6"
    />
    <div
      className="react-bootstrap-table-pagination-list col-md-6 col-xs-6 col-sm-6 col-lg-6 react-bootstrap-table-pagination-list-hidden"
    >
      <ul
        className="pagination react-bootstrap-table-page-btns-ul"
      >
        <li
          className="active page-item"
          title="1"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
          >
            1
          </a>
        </li>
      </ul>
    </div>
  </div>,
]
`;
